libs/mongodb-client/src/lib.rs

```diff
@@ -1,17 +1,291 @@
+mod error;
+
+pub use error::*;
+
+use std::str::FromStr;
+
 use mongodb::{
-    error::Result,
     options::{ClientOptions, ResolverConfig},
     Client,
 };
 
 /// A wrapper to create a new MongoDB client. Please remove me when we do not
 /// need special setup anymore for this.
-pub async fn create(connection_string: impl AsRef<str>) -> Result<Client> {
+pub async fn create(connection_string: impl AsRef<str>) -> Result<Client, Error> {
     let options = if cfg!(target_os = "windows") {
         ClientOptions::parse_with_resolver_config(connection_string, ResolverConfig::cloudflare()).await?
     } else {
         ClientOptions::parse(connection_string).await?
     };
 
-    Client::with_options(options)
+    Ok(Client::with_options(options)?)
+}
+
+/// The parts we need taken from `mongodb` private functions. Please remove me
+/// when they make these apis public.
+pub struct MongoConnectionString {
```

Comment:
Nitpick, but it's not completely clear whether we can remove everything, or everything but the tests, etc. Maybe a separate file? (sorry, it's really a nitpick, feel free to ignore)

Reply:
We can also remove the error module when that's upstream, right?

Reply:
Yep.